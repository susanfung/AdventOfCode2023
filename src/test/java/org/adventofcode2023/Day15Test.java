package org.adventofcode2023;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Day15Test {
    @Test
    void sumOfInitializationSequence() {
        List<String> initializationSequence = Arrays.asList("rn=1", "cm-", "qp=3", "cm=2", "qp-", "pc=4", "ot=9", "ab=5", "pc-", "pc=6", "ot=7");
        Approvals.verify(Day15.sumOfInitializationSequence(initializationSequence));
    }

    @Test
    void hashAlgorithm() {
        String string = "HASH";
        Approvals.verify(Day15.hashAlgorithm(string));
    }
}