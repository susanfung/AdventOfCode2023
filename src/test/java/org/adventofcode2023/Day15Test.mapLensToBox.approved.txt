0 : [[rn, 1], [cm, 2]] 
1 : [] 
3 : [[ot, 7], [ab, 5], [pc, 6]] 
